wsproto/frame_protocol.py

```diff
@@ -142,6 +142,28 @@ def _consume_exactly(self, nbytes):
             yield
         return (yield from self._consume_at_most(nbytes))
 
+    def _parse_extended_payload_length(self, opcode, payload_len):
+        if opcode.iscontrol() and payload_len > 125:
+            raise ParseFailed("Control frame with payload len > 125")
+        if payload_len == 126:
+            data = yield from self._consume_exactly(2)
+            (payload_len,) = struct.unpack("!H", data)
+            if payload_len <= 125:
+                raise ParseFailed(
+                    "Payload length used 2 bytes when 1 would have sufficed")
+        elif payload_len == 127:
+            data = yield from self._consume_exactly(8)
+            (payload_len,) = struct.unpack("!Q", data)
+            if payload_len < 2 ** 16:
```

Comment:
And in fact it does. Here's the output from `dis.dis` on this code, truncated to just cover line 157:

```
157         118 LOAD_FAST                2 (payload_len)
            120 LOAD_CONST              14 (65536)
            122 COMPARE_OP               0 (<)
            124 POP_JUMP_IF_FALSE      134
```

As you can see, something (probably Python's peephole optimiser) spots this static mathematical operation and compiles it into a single constant. So I don't think we need to worry about the conversion. =)